services/system/AuthEcSys/src/RAuthEcSys.cpp

```diff
@@ -9,16 +10,45 @@
 #include <string_view>
 
 using namespace psibase;
+using std::optional;
 using std::string;
 using std::string_view;
 
 namespace SystemService
 {
+
+   struct AuthEcQuery
+   {
+      auto accWithKey(psibase::PublicKey key) const
+      {
+         auto idx = AuthEcSys::Tables{AuthEcSys::service}
+                        .open<AuthEcSys::AuthTable>()
+                        .getIndex<1>()
+                        .subindex(key);
+
+         optional<std::tuple<AccountNumber>> gt;
```

Comment:
Make these args of `accWithKey`

Reply:
But without tuple or the user side of the query gets icky. Unfortunately you'll have to translate between `optional<AccountNumber>` and `optional<tuple<AccountNumber>>`